legate/core/__init__.py

```diff
@@ -14,12 +14,33 @@
 #
 from __future__ import annotations
 
-from ..rc import check_legion
 from ..util.args import parse_library_command_args
 
-check_legion()
+from legion_cffi import is_legion_python, ffi, lib as legion
 
-from legion_cffi import ffi, lib as legion
+if is_legion_python == False:
+    from legion_top import (
+        legion_canonical_python_main,
+        legion_canonical_python_cleanup,
+    )
+    from ..driver.main import prepare_driver, CanonicalDriver
+    import atexit, sys, os
+
+    # parse legate args into legion args
+    sys_argv = [
+        "python",
+    ] + sys.argv
+    driver = prepare_driver(sys_argv, CanonicalDriver)
+    if driver.dry_run:
+        sys.exit(0)
+    legate_argv = driver.cmd
+    legate_env = driver.env
+    # set env
+    for key, value in legate_env.items():
+        if driver.launcher.is_launcher_var(key):
+            os.environ[key] = value
```

Comment:
Did `os.environ = legate_env` not work?

Reply:
No, we can set set `os.environ` like this, its type is `<class 'os._Environ'>` even though it behaves like a dict. I can not find a way to convert a dict into a `os._Environ`.

Reply:
You might be able to do `os.environ.update(legate_env)`.

Reply:
Yeah, it works. BTW, how do you find the API? I tried in docs.python.org, but did not find it.

Reply:
The answer I would like to give is that I went to the documentation for `os.environ`:

https://docs.python.org/3/library/os.html#os.environ

and I saw that it's documented as a "mapping":

https://docs.python.org/3/glossary.html#term-mapping

and I also know that it's mutable, so I looked at the documentation for the `MutableMapping` collection:

https://docs.python.org/3/library/collections.abc.html#collections-abstract-base-classes

and I saw that it adds the `update` method.

The truth is that I opened a python shell and typed `os.environ.` and then tab-tab'ed to see method completions, and picked what looked appropriate.